### ISSUE TITLE
Downloading test support jars fails needlessly

### ISSUE THREAD

**Issue by keithc-ca** [eclipse/openj9#2440]
See https://ci.eclipse.org/openj9/job/PullRequest-Sanity-JDK10-win_x86-64_cmprssptrs-OpenJ9/175/execution/node/117/log/ where this happens:
```
09:58:08 C:/Users/jenkins/workspace/PullRequest-Sanity-JDK10-win_x86-64_cmprssptrs-OpenJ9/openj9/test/\\TestConfig\\lib/jaxb-api.jar exits, skip downloading 
09:58:08 Expected sha1 is: 61dab99f547e2110e42e35f659d9ba27bd00108c, 
09:58:08 Actual sha1 is  : 99f802e0cb3e953ba3d6e698795c4aeb98d37c48. 
09:58:08 Please delete C:/Users/jenkins/workspace/PullRequest-Sanity-JDK10-win_x86-64_cmprssptrs-OpenJ9/openj9/test/\\TestConfig\\lib/jaxb-api.jar and rerun the program!make: *** [makefile:76: getdependency] Error 255
```
If the file does not exist or doesn't have the expected hash, the file should be downloaded. Expecting manual intervention isn't reasonable.

**Comment by llxia** [comment]
- Our script should skip check on windows, but if statement has problem
- It fails now because the newly added jaxb-api.jar sha1 is different on wins

**Comment by keithc-ca** [comment]
The change in #2444 doesn't address this issue. If a file is damaged or the script is updated so a different version is expect, it will not (re-)download the file to fix the situation automatically - manual intervention will still be required.

**Comment by llxia** [comment]
> If a file is damaged or the script is updated so a different version is expect, it will not (re-)download the file to fix the situation automatically

- The current behavior is that:
 - if not jars, the script will download them
 - jars exist, then the script checks for sha1 to verify they are correct jars

By default `wget` will try 20 times if there is exception of fatal errors like “connection refused” or “not found” (404). https://www.gnu.org/software/wget/manual/wget.html
However, the script could be more robust to add re-download if jars are not correct (i.e., different version is expect).
And I think we should add remove jars into `make clean`, so user doesn't need to manually delete all jars. 

@renfeiw when you have time, could you take care of this issue? Thanks.

**Comment by keithc-ca** [comment]
My proposal in #2454 removes the special cases for Windows and for asmtools.jar. I don't know the history of why those exceptions were added to the script, so perhaps they need to be restored (but I hope not).

**Comment by llxia** [comment]
A jar will have different sha1 on windows with cygwin and without cygwin. Since we run this script in AdoptOpenJDK (without cygwin) and OpenJ9 (cygwin), we have decided to skip the check on wins.

There is an issue opened for updating AdoptOpenJDK windows configuration, but it does not seem to happen any time soon.

**Comment by keithc-ca** [comment]
Which jar is _expected_ to be different on Windows? They're each sourced from a unique URL.

The documentation for `Digest::SHA` [1] says the algorithm name should be supplied to `new()`: perhaps the default algorithm is different with and without cygwin? This change makes that argument explicit which I hope will eliminate any such differences.

[1] https://perldoc.perl.org/Digest/SHA.html

**Comment by llxia** [comment]
I think this is general issue, not specific to a particular jar.
Also, skipping wins is a temporary solution. Eventually, wins machines in AdoptOpenJDK should be updated to have cygwin as default (AdoptOpenJDK/openjdk-infrastructure#378).

**Comment by keithc-ca** [comment]
Fixed via #2454.
